### PR TITLE
nutanix improvement

### PR DESCRIPTION
**Pull request by tuxtof**
What this PR does / why we need it:

This PR significantly improves the support of the Nutanix Cloud Platform inside the image-builder:

- based on Nutanix Packer plugin v0.3.1
- directly download OS images if available
- implement dedicated goss preset
- add support of the following OS: Ubuntu-2204, Rocker-8, Rocker-9, Flatcar, Windows 2022
- create a dedicated Nutanix task for ansible provider Role
- add Nutanix CSI dependency inside Linux images
- various fixes
- updated documentation
- add Nutanix OWNERS/OWNERS_ALIASES

Tested with a matrix build for each OS /  k8s supported version

image

### CODE REVIEW

**Comment by tuxtof** [comment]
> /retest

seems OK now, did you change something in the backend @mboersma ?

**Review by mboersma — COMMENTED** [review]
/lgtm

@tuxtof I didn't change anything. The failures looked unrelated to your changes, and were apparently flakes. :-|

**Comment by tuxtof** [comment]
> Does this replace #985?

yes

**Comment by kkeshavamurthy** [comment]
Looks good @tuxtof 
Can you pls squash the commits.

**Comment by tuxtof** [comment]
> Looks good @tuxtof
> Can you pls squash the commits.

@kkeshavamurthy done commits squashed

File: images/capi/packer/nutanix/ubuntu-2004.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "build_name": "ubuntu-2004",
   "distro_name": "ubuntu",
   "guest_os_type": "Linux",
+  "image_url": "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img",
```
Comment (deepakm-ntnx) [inline images/capi/packer/nutanix/ubuntu-2004.json:5]: can we use specific image of 2004 instead of taking latest version?

Comment (tuxtof) [reply images/capi/packer/nutanix/ubuntu-2004.json:5]: image-builder automatically launch a yum udpate in the image so you always finish with the most recent up to date version.
Using a sepcific image is therefore useless , the latest allow to improve the building time.

**Pull request merged**